### PR TITLE
fix(storage): react-native picks up wrong utils/client/runtime

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes

Added appropriate `react-native` root fields to ensure the module resolution picks up correct code path of `packages/storage` in a react-native app.

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/storage/package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
 	"description": "Storage category of aws-amplify",
 	"main": "./lib/index.js",
 	"module": "./lib-esm/index.js",
+	"react-native": "./lib-esm/index.js",
```
Comment (AllanZhengYP) [inline packages/storage/package.json:7]: Why is this needed?

**Review by jimblanc — APPROVED** [review]

**Review by cshfang — APPROVED** [review]

File: packages/storage/package.json (hunk 1 above)
Comment (HuiSF) [reply packages/storage/package.json:7]: metro picks up `lib` without this.

**Review by kvramyasri7 — APPROVED** [review]
Thanks @HuiSF

**Review by AllanZhengYP — APPROVED** [review]
Thanks a lot! @HuiSF
